addons/httprox/common/src/main/java/org/commonjava/indy/httprox/util/CertUtils.java

```diff
@@ -0,0 +1,196 @@
+package org.commonjava.indy.httprox.util;
+
+//import sun.security.tools.keytool.CertAndKeyGen;
+
+import sun.security.x509.AlgorithmId;
+import sun.security.x509.BasicConstraintsExtension;
+import sun.security.x509.CertificateAlgorithmId;
+import sun.security.x509.CertificateExtensions;
+import sun.security.x509.CertificateSerialNumber;
+import sun.security.x509.CertificateValidity;
+import sun.security.x509.CertificateVersion;
+import sun.security.x509.CertificateX509Key;
+import sun.security.x509.X500Name;
+import sun.security.x509.X509CertImpl;
+import sun.security.x509.X509CertInfo;
```

Comment:
Have you tried this on OpenJDK yet? We might need to use the code in jhttpc for certificate handling instead...I'm not sure these classes are safe to use. They might not be considered part of the API, and may not be supported in the future.

Reply:
This is the code I googled and worked. I agree we should change to OpenJDK later to avoid proprietary lib.